Lib/test/test_inspect.py

```diff
@@ -202,6 +202,31 @@ def test_iscoroutine(self):
                     gen_coroutine_function_example))))
         self.assertTrue(inspect.isgenerator(gen_coro))
 
+        # Use subtest initially to see both failures.
+        with self.subTest("Wrapper not recognised."):
+            # First case: sync function returning an awaitable.
+            async def _fn3():
+                pass
+
+            def fn3():
+                return _fn3()
+
+            # TODO: Move this to decorator function.
+            fn3.__code__ = fn3.__code__.replace(
+                co_flags=fn3.__code__.co_flags | inspect.CO_COROUTINE
+            )
+
+            self.assertTrue(inspect.iscoroutinefunction(fn3))
+
+        with self.subTest("Awaitable instance not recongnised."):
+            # Second case: a class with an async def __call__.
+            # - instance is awaitable.
+            class Cl:
+                async def __call__(self):
```

Comment:
In the future don't forget to test objects with `.__func__` set, because of how `_has_code_flag` works :)

Reply:
I haven't followed exactly what you mean here. Sorry. 🤔

Reply:
Oh, sorry. I've meant that objects like `staticmethod` and `classmethod` have `__func__` property set, where the original function is stored. `inspect` generally supports this pattern and it should be tested in this case as well :)

Reply:
@sobolevn — OK, let me have a play and see. Thanks!